src/main/java/org/spongepowered/api/CatalogType.java

```diff
@@ -60,5 +64,13 @@
      * @return The human-readable name of this catalog type
      */
     String getName();
+    
+    @Override
+    default Text toText() {
```

Comment:
Yes, we had (See https://github.com/SpongePowered/SpongeAPI/pull/808#discussion_r49005018). Thats why i explicitly made this three levelled approach.

I removed it from that PR since it was a separate problem of the current text API and I didn't want to push everything in there.

I have since thought about it a lot.

> Convince me why **every** `CatalogType` needs to be `TextRepresentable`.

`CatalogType`s have a name. Not using it for displaying it to the user is just wrong, this should be prevented escpecially if the `toString()` fallback is unusable for displaying in chat.
````
toString: BlockColored{id=minecraft:stained_hardened_clay, name=minecraft:stained_hardened_clay, translation=SpongeTranslation{id=tile.clayHardenedStained.name}, toString=BlockColored{id=minecraft:stained_hardened_clay, name=minecraft:stained_hardened_clay, translation=SpongeTranslation{id=tile.clayHardenedStained.name}, class=net.minecraft.block.BlockColored}}

toText: Stained Clay
````

IMO everything that has a name should be forced to display it in chat unless it has something better to display (custom name etc).
I just don't want to go the road down to a interface like this:

````java
public interface Named extends TextRepresentable {

    String getName();

    default Text toText() {
        return Text.of(getName());
    }

}
````

Default implementations can be overwritten and don't impose anything on implementations except their existance.

I'm not sure about the current implementation though.
(I though about moving it to a `Function<CatalogType,Text>`)